ella_galleries/views.py

```diff
@@ -33,7 +36,13 @@ def gallery_item_detail(request, context, item_slug=None, url_remainder=None):
         try:
             item = item_sorted_dict[item_slug]
         except KeyError:
-            raise Http404()
+            # check if flag is set then 301 to the main gallery url else 404 since slug was not found
+            GALLERY_REDIRECT_ENABLED = getattr(settings, 'GALLERY_REDIRECT_ENABLED', False)
+            if GALLERY_REDIRECT_ENABLED:
+                redirect_url =  request.path_info[0:request.path_info.find('/item/')+1] 
+                return redirect(redirect_url, permanent=True)
+            else:    
```

Comment:
also fwiw you should redirect to `gallery.get_absolute_url()` and not based on potential arbitrary url (there could be a category named `item` which would make this code entirely wrong.